src/core/border_router/routing_manager.cpp

```diff
@@ -0,0 +1,799 @@
+    {
+        Ip6::NetworkPrefix randomOnLinkPrefix;
+
+        otLogNoteBr("no valid on-link prefix found in settings, generating new one");
+
+        if (randomOnLinkPrefix.GenerateRandomUla() != OT_ERROR_NONE)
+        {
+            otLogCritBr("failed to generate random on-link prefix");
+            ExitNow(error = OT_ERROR_FAILED);
+        }
+
+        randomOnLinkPrefix.m8[6] = 0;
+        randomOnLinkPrefix.m8[7] = 0;
+        mLocalOnLinkPrefix.Set(randomOnLinkPrefix);
+
+        IgnoreError(Get<Settings>().SaveOnLinkPrefix(mLocalOnLinkPrefix));
+    }
+
+exit:
+    return error;
+}
+
+void RoutingManager::Start(void)
+{
+    if (!mIsRunning)
+    {
+        StartRouterSolicitation();
+        mIsRunning = true;
+    }
+}
+
+void RoutingManager::Stop(void)
+{
+    VerifyOrExit(mIsRunning);
+
+    UnpublishLocalOmrPrefix();
+
+    // Use empty OMR & on-link prefixes to invalidate possible advertised prefixes.
+    SendRouterAdvertisement(nullptr, 0, nullptr);
+
+    memset(mAdvertisedOmrPrefixes, 0, sizeof(mAdvertisedOmrPrefixes));
+    mAdvertisedOmrPrefixNum = 0;
+
+    mAdvertisedOnLinkPrefix = nullptr;
+
+    mDiscoveredOnLinkPrefix.Clear();
+    mDiscoveredOnLinkPrefixInvalidTimer.Stop();
+
+    mRouterAdvertisementTimer.Stop();
+    mRouterAdvertisementCount = 0;
+
+    mRouterSolicitTimer.Stop();
+    mRouterSolicitCount = 0;
+
+    mIsRunning = false;
+
+exit:
+    return;
+}
+
+void RoutingManager::RecvIcmp6Message(uint32_t            aInfraIfIndex,
+                                      const Ip6::Address &aSrcAddress,
+                                      const uint8_t *     aBuffer,
+                                      uint16_t            aBufferLength)
+{
+    const Ip6::Icmp::Header *icmp6Header;
+
+    // The Border Routing function is enabled only when we are attached.
+    VerifyOrExit(IsInitialized() && Get<Mle::MleRouter>().IsAttached());
+
+    VerifyOrExit(aInfraIfIndex == mInfraIfIndex);
+    VerifyOrExit(aBuffer != nullptr && aBufferLength >= sizeof(*icmp6Header));
+
+    icmp6Header = reinterpret_cast<const Ip6::Icmp::Header *>(aBuffer);
+
+    switch (icmp6Header->GetType())
+    {
+    case Ip6::Icmp::Header::kTypeRouterAdvert:
+        HandleRouterAdvertisement(aSrcAddress, aBuffer, aBufferLength);
+        break;
+    case Ip6::Icmp::Header::kTypeRouterSolicit:
+        HandleRouterSolicit(aSrcAddress, aBuffer, aBufferLength);
+        break;
+    default:
+        ExitNow();
+    }
+
+exit:
+    return;
+}
+
+void RoutingManager::HandleNotifierEvents(Events aEvents)
+{
+    VerifyOrExit(IsInitialized());
+
+    if (aEvents.Contains(kEventThreadRoleChanged))
+    {
+        if (Get<Mle::MleRouter>().IsAttached())
+        {
+            Start();
+        }
+        else
+        {
+            Stop();
+        }
+    }
+
+    if (aEvents.Contains(kEventThreadNetdataChanged))
+    {
+        if (mIsRunning)
+        {
+            EvaluateRoutingPolicy();
+        }
```

Comment:
```suggestion
        EvaluateRoutingPolicy();
```
Check `mIsRunning` in `EvaluateRoutingPolicy` ?